consensus/core/src/authority_service.rs

```diff
@@ -212,13 +214,38 @@ impl<C: CoreThreadDispatcher> NetworkService for AuthorityService<C> {
         // For now ask dag state directly
         let blocks = self.dag_state.read().get_blocks(&block_refs);
 
-        // Return the serialised blocks
-        let result = blocks
+        // Now check if an ancestor's round is higher than the one that the peer has. If yes, then serve
+        // that ancestor blocks up to 10
+        let all_ancestors = blocks
+            .iter()
+            .flatten()
+            .flat_map(|block| block.ancestors().to_vec())
+            .filter(|block_ref| highest_accepted_rounds[block_ref.author] < block_ref.round)
```

Comment:
I don't see length of `highest_accepted_rounds` validated before. But I think highest_accepted_rounds should indeed remain optional and allowed to be empty, because there are good use cases for fetching only blocks with specified refs. Fetching ancestor blocks can be guarded when highest_accepted_rounds matches the committee size.

Reply:
hhm I would swear that I had this checked - thanks for noticing @mwtian ! Yes, I'll make it either optional (empty) or otherwise should match the committee size